internal/operators/ocs/validations.go

```diff
@@ -44,35 +27,50 @@ type Config struct {
 	OCSDeploymentType              string `envconfig:"OCS_DEPLOYMENT_TYPE" default:"None"`
 }
 
-// ValidateOCSRequirements is used to validate min requirements of OCS
-func (o *ocsValidator) ValidateRequirements(cluster *models.Cluster) (api.ValidationStatus, string) {
+type resources struct {
+	host              *models.Host
+	cpuCount          *int64
+	totalRAM          *int64
+	diskCount         *int64
+	hostsWithDisks    *int64
+	insufficientHosts *[]string
+}
+
+// validateRequirements is used to validate min requirements of OCS
+func (o *operator) validateRequirements(cluster *models.Cluster) (api.ValidationStatus, string) {
 	var status string
-	var err error
-	var inventoryMissing bool
 	hosts := cluster.Hosts
+	numAvailableHosts := len(hosts)
 
-	if int64(len(hosts)) < o.OCSRequiredHosts {
+	//Check if number of hosts is less than required for compact mode
+	if int64(numAvailableHosts) < o.config.OCSRequiredHosts {
 		status = "Insufficient hosts to deploy OCS. A minimum of 3 hosts is required to deploy OCS. "
 		o.log.Info("Validate Requirements status ", status)
 		return api.Failure, status
 	}
 	var cpuCount int64 = 0       //count the total CPUs on the cluster
 	var totalRAM int64 = 0       // count the total available RAM on the cluster
 	var diskCount int64 = 0      // count the total disks on all the hosts
-	var hostsWithDisks int64 = 0 // to determine total number of hosts with disks. OCS requires atleast 3 hosts with non-bootable disks
+	var hostsWithDisks int64 = 0 // to determine total number of hosts with disks. OCS requires at least 3 hosts with non-bootable disks
 	var insufficientHosts []string
-	if int64(len(hosts)) == o.OCSRequiredHosts { // for only 3 hosts, we need to install OCS in compact mode
+
+	r := resources{
+		cpuCount:          &cpuCount,
+		totalRAM:          &totalRAM,
+		diskCount:         &diskCount,
+		hostsWithDisks:    &hostsWithDisks,
+		insufficientHosts: &insufficientHosts,
+	}
+
+	if int64(numAvailableHosts) == o.config.OCSRequiredHosts { // for only 3 hosts, we need to install OCS in compact mode
 		for _, host := range hosts {
-			inventoryMissing, err = o.nodeResources(host, &cpuCount, &totalRAM, &diskCount, &hostsWithDisks, &insufficientHosts)
+			r.host = host
+			res, err := o.checkInventoryStatus(r)
```

Comment:
As you are modifying resources in other functions, can you pass it as a pointer? Not sure if the values are retained when you exit from the function.

Reply:
Also rename `res` to `status`

Reply:
Done

Reply:
You can pass `r` as it is, I did not notice the `cpu, ram` etc were already passed as a pointer. You can ignore my previous comment on passing it as a pointer.